spyder/plugins/editor/plugin.py

```diff
@@ -1653,13 +1653,13 @@ def update_recent_file_menu(self):
         """Update recent file menu"""
         recent_files = []
         for fname in self.recent_files:
-            if self.is_file_opened(fname) is None and osp.isfile(fname):
+            if osp.isfile(fname):
                 recent_files.append(fname)
         self.recent_file_menu.clear()
         if recent_files:
             for fname in recent_files:
                 action = create_action(self, fname,
-                                       icon=ima.icon('FileIcon'),
+                                       icon=ima.get_icon_by_extension(fname, scale_factor=1.0),
                                        triggered=self.load)

```

Comment:
To fix the Pep8speaks warning, please change this code to be

```python
action = create_action(
    self, fname,
    icon=ima.get_icon_by_extension(fname, scale_factor=1.0),
    triggered=self.load)
```

Reply:
@ccordoba12 done!